services/certificate/certificate-expiration.service.js

```diff
@@ -0,0 +1,110 @@
+import emojic from 'emojic'
+import Joi from 'joi'
+import { BaseService, pathParam, queryParam } from '../index.js'
+import trace from '../../core/base-service/trace.js'
+
+const queryParamSchema = Joi.object({
+  warningDays: Joi.number().min(0),
+  dangerDays: Joi.number().min(0),
+}).required()
```

Comment:
In general, we don't provide this sort of customisation via URL params.
I think we should make a decision on some thresholds for colours and just apply it.
There's some more comments on formatting further down.

Reply:
I disagree about this:
1. Something that makes sense for LetsEncrypt (3 months) will not make sense for a manually renewed cert that lasts 12 months.
2. It would make testing the colors very unreliable (or at least very tricky)

Reply:
I think we can ignore "it would be hard to test" as a consideration. It is fine for the live tests to omit the colour. If we switch to using `colorScale()`, that function is already well tested. If we implement custom formatting logic that we want to test, we can either pull that out to a method and unit test the method in a .`spec.js`, or you could do integration tests with mocks. There are ways we can handle that. We probably don't want to be leaning too hard on live tests with a service like this that doesn't depend on a single upstream service anyway.

If what is "good" or "bad" is so context dependent that we can't possibly define it, I think my suggestion is:

- The expiry date is already expired: red - that is unambiguously bad in 100% of cases
- The expiry date is in the future: informational blue